kotsadm/pkg/handlers/license.go

```diff
@@ -109,31 +121,100 @@ func SyncLicense(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	entitlements, expiresAt, err := GetLicenseEntitlements(latestLicense)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(500)
+		return
+	}
+
 	syncLicenseResponse := SyncLicenseResponse{
 		ID:              latestLicense.Spec.LicenseID,
 		ChannelName:     latestLicense.Spec.ChannelName,
 		LicenseSequence: latestLicense.Spec.LicenseSequence,
 		LicenseType:     latestLicense.Spec.LicenseType,
-		Entitlements:    []EntitlementResponse{},
+		Entitlements:    entitlements,
+		ExpiresAt:       expiresAt,
+	}
+
+	JSON(w, 200, syncLicenseResponse)
+}
+
+func GetLicense(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Headers", "content-type, origin, accept, authorization")
+
+	if r.Method == "OPTIONS" {
+		w.WriteHeader(200)
+		return
+	}
+
+	sess, err := session.Parse(r.Header.Get("Authorization"))
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(500)
+		return
+	}
+
+	// we don't currently have roles, all valid tokens are valid sessions
+	if sess == nil || sess.ID == "" {
+		w.WriteHeader(401)
+		return
+	}
+
+	appSlug := mux.Vars(r)["appSlug"]
+	foundApp, err := app.GetFromSlug(appSlug)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(500)
+		return
 	}
 
-	for key, entititlement := range latestLicense.Spec.Entitlements {
+	license, err := kotsadmlicense.Get(foundApp.ID)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(500)
+		return
+	}
+
+	entitlements, expiresAt, err := GetLicenseEntitlements(license)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(500)
+		return
+	}
+
+	getLicenseResponse := GetLicenseResponse{
+		ID:              license.Spec.LicenseID,
+		ChannelName:     license.Spec.ChannelName,
+		LicenseSequence: license.Spec.LicenseSequence,
+		LicenseType:     license.Spec.LicenseType,
+		Entitlements:    entitlements,
+		ExpiresAt:       expiresAt,
+	}
+
+	JSON(w, 200, getLicenseResponse)
+}
+
+func GetLicenseEntitlements(license *kotsv1beta1.License) ([]EntitlementResponse, time.Time, error) {
```

Comment:
this does not need to be exported
```suggestion
func getLicenseEntitlements(license *kotsv1beta1.License) ([]EntitlementResponse, time.Time, error) {
```